Source/WebCore/editing/Editor.cpp

```diff
@@ -4381,14 +4382,18 @@ void Editor::registerAttachments(Vector<SerializedAttachmentData>&& data)
         client->registerAttachments(WTFMove(data));
 }
 
-void Editor::registerAttachmentIdentifier(const String& identifier, const HTMLImageElement& imageElement)
+void Editor::registerAttachmentIdentifier(const String& identifier, const AttachmentAssociatedElement& element)
 {
     auto* client = this->client();
     if (!client)
         return;
 
     auto attachmentInfo = [&]() -> std::optional<std::tuple<String, String, Ref<FragmentedSharedBuffer>>> {
-        auto* renderer = dynamicDowncast<RenderImage>(imageElement.renderer());
+        RefPtr imageElement = dynamicDowncast<HTMLImageElement>(element.asHTMLElement());
+        if (!imageElement)
+            return std::nullopt;
+
+        auto* renderer = dynamicDowncast<RenderImage>(imageElement->renderer());
```

Comment:
Nit - maybe we can turn this into `CheckedPtr` while we're here?

Reply:
Done.